tests/utils/exec.ts

```diff
@@ -629,171 +613,29 @@ async function setupTestAgent(privateKeyPem: PrivateKeyPem, logger: Logger) {
-  );
-  const polykeyPath = path.resolve(
-    path.join(globalThis.projectDir, 'src/bin/polykey.ts'),
-  );
-  return new Promise((resolve, reject) => {
-    child_process.execFile(
-      'nsenter',
-      [
-        ...nsenter(usrnsPid, netnsPid),
-        'ts-node',
-        '--project',
-        tsConfigPath,
-        polykeyPath,
-        ...args,
-      ],
-      {
-        env,
-        cwd,
-        windowsHide: true,
-      },
-      (error, stdout, stderr) => {
-        if (error != null && error.code === undefined) {
-          // This can only happen when the command is killed
-          return reject(error);
-        } else {
-          // Success and Unsuccessful exits are valid here
-          return resolve({
-            exitCode: error && error.code != null ? error.code : 0,
-            stdout,
-            stderr,
-          });
-        }
-      },
-    );
-  });
-}
-
-/**
- * Launch pk command through subprocess inside a network namespace
- * This is used when a subprocess functionality needs to be used
- * This is intended for non-terminating subprocesses
- * @param env Augments env for command execution
- * @param cwd Defaults to temporary directory
- */
-async function pkSpawnNs(
-  usrnsPid: number,
-  netnsPid: number,
-  args: Array<string> = [],
-  env: Record<string, string | undefined> = {},
-  cwd?: string,
-  logger: Logger = new Logger(pkSpawnNs.name),
-): Promise<ChildProcess> {
-  cwd =
-    cwd ?? (await fs.promises.mkdtemp(path.join(os.tmpdir(), 'polykey-test-')));
-  env = {
-    ...process.env,
-    ...env,
-  };
-  // Recall that we attempt to connect to all specified seed nodes on agent start.
-  // Therefore, for testing purposes only, we default the seed nodes as empty
-  // (if not defined in the env) to ensure no attempted connections. A regular
-  // PolykeyAgent is expected to initially connect to the mainnet seed nodes
-  env['PK_SEED_NODES'] = env['PK_SEED_NODES'] ?? '';
-  const tsConfigPath = path.resolve(
-    path.join(globalThis.projectDir, 'tsconfig.json'),
-  );
-  const polykeyPath = path.resolve(
-    path.join(globalThis.projectDir, 'src/bin/polykey.ts'),
-  );
-  const subprocess = child_process.spawn(
-    'nsenter',
-    [
-      ...nsenter(usrnsPid, netnsPid),
-      'ts-node',
-      '--project',
-      tsConfigPath,
-      polykeyPath,
-      ...args,
-    ],
-    {
-      env,
-      cwd,
-      stdio: ['pipe', 'pipe', 'pipe'],
-      windowsHide: true,
-      shell: true,
-    },
-  );
-  const rlErr = readline.createInterface(subprocess.stderr!);
-  rlErr.on('line', (l) => {
-    // The readline library will trim newlines
-    logger.info(l);
-  });
-  return subprocess;
+function escapeShellArgs(arg: string): string {
+  return arg.replace(/(["\s'$`\\])/g, '\\$1');
 }
 
 export {
-  exec,
-  pk,
-  pkStdio,
+  polykeyPath,
   pkExec,
   pkSpawn,
-  pkStdioTarget,
-  pkExecTarget,
-  pkSpawnTarget,
   pkExpect,
+  pkStdio,
+  exec,
+  spawn,
+  pk,
+  setupTestAgent,
```

Comment:
See if you can move this to `tests/utils/utils.ts`.

Reply:
`setupTestAgent` has been moved.